ui/src/app/shared/components/graph/graph-panel.tsx

```diff
@@ -72,13 +72,23 @@ export const GraphPanel = (props: Props) => {
 
     const visible = (id: Node) => {
         const label = props.graph.nodes.get(id);
-        return (
-            (nodeGenres[label.genre] &&
-                (!nodeClassNames || Object.entries(nodeClassNames).find(([className, checked]) => checked && (label.classNames || '').split(' ').includes(className))) &&
-                (!nodeTags || Object.entries(nodeTags).find(([tag, checked]) => !label.tags || (checked && label.tags.has(tag)))) &&
-                !nodeSearchKeyword) ||
-            label.label.includes(nodeSearchKeyword)
```

Comment:
This line broke the filter. If `nodeSearchKeyword` was empty it would match all nodes, rendering the filter moot